src/main/kotlin/org/neo4j/graphql/GraphQLExtensions.kt

```diff
@@ -49,9 +49,13 @@ fun GraphQLFieldsContainer.relationshipFor(name: String): RelationshipInfo? {
             ?: throw IllegalArgumentException("$name is not defined on ${this.name}")
     val fieldObjectType = field.type.inner() as? GraphQLFieldsContainer ?: return null
 
-    val (relDirective, isRelFromType) = if (isRelationType()) {
+    val (relDirective, inverse) = if (isRelationType()) {
+        val typeName = this.name
         (this as? GraphQLDirectiveContainer)
-            ?.getDirective(DirectiveConstants.RELATION)?.let { it to false }
+            ?.getDirective(DirectiveConstants.RELATION)?.let {
+                // do inverse mapping, if the current type  is the `to` mapping of the relation
+                it to (fieldObjectType.getFieldDefinition(it.getArgument(RELATION_TO, null))?.name == typeName)
```

Comment:
we need to be careful, this doesn't always work. if you have a rich relationship which has the same type in to/from then we need to use the explicit information from that type.

Reply:
we need the information the rel-field then I mean that indicates the correct direction otherwise it's "random" if from/to is matching the type name.

Reply:
see: https://github.com/neo4j-graphql/neo4j-graphql-java/issues/45